app/models/katello/concerns/setting_extensions.rb

```diff
@@ -6,8 +6,11 @@ module SettingExtensions
       included do
         validates :value, inclusion: { in: ::Katello::RootRepository::DOWNLOAD_POLICIES },
           if: ->(setting) { setting.name == 'default_download_policy' }
+        validates_with ::Katello::Validators::ContentDefaultHttpProxySettingValidator,
```

Comment:
I found that calling the validator from here actually works. @ezr-ondrej hopefully that's ..valid? :yum:

Reply:
It is "valid" but there is a `validates_with` setting DSL: https://github.com/theforeman/foreman/blob/develop/developer_docs/how_to_create_a_plugin.asciidoc#validate-setting-values

in a long run, I think you should not need patching the setting class :)

Reply:
For some reason it wasn't working when I tried to call `validates_with` from the `settings do` block.

Our setting class was already patched which also allows the Rails after-actions, does the DSL have a solution for this too?

Reply:
(I'm talking about this: )
```rb
after_save :recalculate_errata_status
after_save :add_organizations_and_locations_if_global_http_proxy
after_commit :update_global_proxies
```

Reply:
Yeah, that's why I mentioned in a long run... these are unfortunatelly not possible ATM through DSL, but it is definitelly a great idea to add and is not so hard IMHO :)